src/sqlancer/GlobalState.java

```diff
@@ -118,6 +120,93 @@ public boolean executeStatement(Query q) throws SQLException {
         return success;
     }
 
+    public ResultSet executeStatementAndGet(Query q) throws SQLException {
```

Comment:
It seems that much code is duplicated here with respect to the `executeStatement` method. Could you try to factor out common functionality?